services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/MultipartUtils.java

```diff
@@ -35,6 +37,8 @@ public static String getBucket(StorageLocationSetting storageLocationSetting) {
 
 	/** Create a new S3 key given a file name and storage location settings. */
 	public static String createNewKey(String userId, String fileName, StorageLocationSetting storageLocationSetting) {
+		// S3 has token signing issues when the key contains non-ascii characters
+		NameValidation.validateName(fileName);

```

Comment:
Consider double checking if the baseKey of the potential storage location is validated as well (e.g. BaseKeyStorageLocationProcessor)